app/src/main/java/de/westnordost/streetcomplete/quests/crossing_signals/AddCrossingSignals.kt

```diff
@@ -0,0 +1,62 @@
+package de.westnordost.streetcomplete.quests.crossing_signals
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.osm.geometry.ElementGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmElementQuestType
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.PEDESTRIAN
+import de.westnordost.streetcomplete.osm.Tags
+import de.westnordost.streetcomplete.osm.isCrossing
+import de.westnordost.streetcomplete.osm.updateCheckDateForKey
+import de.westnordost.streetcomplete.osm.updateWithCheckDate
+import de.westnordost.streetcomplete.quests.YesNoQuestForm
+import de.westnordost.streetcomplete.util.ktx.toYesNo
+
+class AddCrossingSignals : OsmElementQuestType<Boolean> {
+
+    private val crossingFilter by lazy { """
+        nodes with
+          highway = crossing
+          and foot != no
+          and !crossing:signals
+          and (!crossing or crossing = island)
+    """.toElementFilterExpression() }
+    /* only looking for crossings that have no crossing=* at all set because if the crossing had
+     * signals, it would have been tagged with crossing=traffic_signals */
+
+    private val excludedWaysFilter by lazy { """
+        ways with
+          highway and access ~ private|no
+          or highway = service
```

Comment:
FYI, this would exclude things like entrances to shopping malls and some residential subdivisions at signalized intersections, for example:

https://www.openstreetmap.org/node/4146380881
https://www.openstreetmap.org/node/5860157357
https://www.openstreetmap.org/node/11419081726

However, I can see how such crossings would get very spammy because of the legacy of openstreetmap/iD#8463. Do you have a way of checking whether the crossing is one node away from a `highway=traffic_signals` node? I think that would be a much better indicator of whether signalization is likely or not.

Reply:
You mention the legacy of that iD issue, but this hasn't been the reason for excluding `highway=service`, but rather the deliberation that

1. many service roads that are actually driveways are not (yet) tagged with `service=driveway`
2. it is generally really unlikely that a crossing with a service road is signalized. 

My gut estimation is that it would be too spammy. That's because the crossing island one is also not asked for any `highway=service`. Yeah, that's all gut estimation, but to topple that estimation, an argument backed with some numbers would be necessary (and I would say it is hard to get this information).

Point 1 is a big a problem. It would be much less spammy if this could be classified as solved, for example by a quest that asks for the type of service road, but IIRC that attempt failed because we determined that there are cases where the type of service road cannot be defined any more detailed than `highway=service` (i.e. no sub-type).

> Do you have a way of checking whether the crossing is one node away from a highway=traffic_signals node?

It is somehow possible, but it would be very complex because in the data structure, the vertices of ways do not link back to the way(s) they are contained within. Too complex to find out this detail, it also covers only a very small number of cases: Those cases where someone mapped the vehicular traffic lights but did forget to add the crossing type.

Reply:
> Those cases where someone mapped the vehicular traffic lights but did forget to add the crossing type.

That’s fair, most `crossing:signals` ([69.67% of them](https://taginfo.openstreetmap.org/keys/crossing:signals#combinations)) are already being combined with `crossing:markings`, although the proportion will decrease over time thanks to this quest.